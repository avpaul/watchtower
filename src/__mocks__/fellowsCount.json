[
  {
    "name": "Week 1",
    "Off Track": 10,
    "On Track": 30,
    "PIP": 3
  },
  {
    "name": "Week 2",
    "Off Track": 12,
    "On Track": 20,
    "PIP": 2
  },
  {
    "name": "Week 3",
    "Off Track": 8,
    "On Track": 22,
    "PIP": 4
  },
  {
    "name": "Week 4",
    "Off Track": 12,
    "On Track": 19,
    "PIP": 5
  },
  {
    "name": "Week 5",
    "Off Track": 4,
    "On Track": 16,
    "PIP": 6
  },
  {
    "name": "Week 6",
    "Off Track": 3,
    "On Track": 23,
    "PIP": 2
  },
  {
    "name": "Week 7",
    "Off Track": 5,
    "On Track": 22,
    "PIP": 1
  },
  {
    "name": "Week 8",
    "Off Track": 9,
    "On Track": 29,
    "PIP": 9
  },
  {
    "name": "Week 9",
    "Off Track": 3,
    "On Track": 30,
    "PIP": 6
  },
  {
    "name": "Week 10",
    "Off Track": 5,
    "On Track": 25,
    "PIP": 0
  },
  {
    "name": "Week 11",
    "Off Track": 6,
    "On Track": 26,
    "PIP": 6
  },
  {
    "name": "Week 12",
    "Off Track": 26,
    "On Track": 30,
    "PIP": 3
  }
]
