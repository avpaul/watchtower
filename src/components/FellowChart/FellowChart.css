.fellow-chart {
  background-color: rgba(4, 89, 228, 0.7);
  padding: 30px 60px 50px;
  position: relative;
}

.fellow-chart-close {
  cursor: pointer;
  position: absolute;
  font-size: 2em;
  top: -4px;
  right: 20px;
  opacity: 1;
}
.base {
  height: 1.5em;
  width: 100%;
  background-color: rgba(4, 89, 228, 0.7);
}
.fellow-chart-close .far.fa-times-circle {
  color: white;
}

.base::before {
  content: ' ';
  width: 0;
  height: 0;
  border-left: 1.5em solid transparent;
  border-right: 1.5em solid transparent;
  border-bottom: 1.5em solid rgba(4, 89, 228, 0.7);
  position: absolute;
  margin-top: -1.5em;
  margin-left: var(--fellow-chart-tooltip);
}
