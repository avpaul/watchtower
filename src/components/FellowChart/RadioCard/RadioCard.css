.radio-card__input .icon::after {
  display: inline-block;
  font-style: normal;
  font-variant: normal;
  text-rendering: auto;
  -webkit-font-smoothing: antialiased;
}

input[type='radio'].radio-card__input {
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.radio-card__input:after {
  width: 20px;
  height: 20px;
  color: rgba(4, 89, 228, 0.7);
  font-family: 'Font Awesome 5 Free';
  font-size: 20px;
  font-weight: 500;
  content: '\f111';
}

.radio-card__input:checked:after {
  width: 20px;
  height: 20px;
  color: rgba(4, 89, 228, 0.7);
  font-family: 'Font Awesome 5 Free';
  font-size: 20px;
  font-weight: 500;
  content: '\f192';
}
.radio-card__input:focus {
  outline: none;
}

.radio-card {
  background-color: #fff;
  border-radius: 4px;
  width: max-content;
  height: 55px;
  margin-right: 20px;
  margin-bottom: 30px;
  padding-right: 29px;
}

.radio-card__input.form-check-input {
  margin: 17.5px 22px;
}
