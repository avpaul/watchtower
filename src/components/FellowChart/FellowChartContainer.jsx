import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import FellowChart from './FellowChart';
import {
  convertHistory,
  mergeHistory
} from '../../services/convertHistoryData';
import Loader from '../Loader/Loader';

export const getFellowsCount = (countSummary, filter) => {
  const d0a = countSummary['D0A Simulations'];
  const d0b = countSummary['D0B Apprenticeship'];

  const all = mergeHistory(d0a, d0b);
  const d0aArray = convertHistory(d0a);
  const d0bArray = convertHistory(d0b);

  const allArray = convertHistory(all);
  const fellowCountArray = { ALL: allArray, D0A: d0aArray, D0B: d0bArray };
  return fellowCountArray[filter];
};

export const getRadioCardData = fellowsCount => {
  const thisWeekData = fellowsCount.find(entry => entry.name === 'Week 12') || {
    'On Track': 0,
    'Off Track': 0,
    PIP: 0
  };
  const countOnTrack = thisWeekData['On Track'];
  const countOffTrack = thisWeekData['Off Track'];
  const countPIP = thisWeekData.PIP;
  return [
    {
      value: 'All',
      name: 'All',
      count: countOnTrack + countOffTrack
    },
    { value: 'On Track', name: 'Fellows On Track', count: countOnTrack },
    { value: 'Off Track', name: 'Fellows Off Track', count: countOffTrack },
    { value: 'PIP', name: 'Fellows On PIP', count: countPIP }
  ];
};
export const getFilterTooltips = filter => {
  const tooltipOptions = {
    ALL: { '--fellow-chart-tooltip': '8%' },
    D0A: { '--fellow-chart-tooltip': '30%' },
    D0B: { '--fellow-chart-tooltip': '52%' }
  };
  return tooltipOptions[filter];
};

const FellowChartContainer = props => {
  const { loading, countSummary, filter, handleChartClose } = props;
  let fellowsCount = [];
  if (countSummary['D0A Simulations']) {
    fellowsCount = getFellowsCount(countSummary, filter);
  }
  const radioCardData = getRadioCardData(fellowsCount);
  const fellowChartTooltip = getFilterTooltips(filter);
  return (
    <Fragment>
      {!loading && fellowsCount.length > 0 ? (
        <FellowChart
          fellowsCount={fellowsCount}
          radioCardOptions={radioCardData}
          handleChartClose={handleChartClose}
          fellowChartTooltip={fellowChartTooltip}
        />
      ) : (
        <Loader />
      )}
    </Fragment>
  );
};

FellowChartContainer.defaultProps = {
  filter: 'ALL',
  countSummary: {}
};

FellowChartContainer.propTypes = {
  handleChartClose: PropTypes.func.isRequired,
  countSummary: PropTypes.shape({
    'D0A Simulations': PropTypes.object,
    'D0B Apprenticeship': PropTypes.object
  }),
  filter: PropTypes.string,
  loading: PropTypes.bool.isRequired
};

export default FellowChartContainer;
