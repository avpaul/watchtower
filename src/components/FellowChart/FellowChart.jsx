import React, { Component } from 'react';
import PropTypes from 'prop-types';

import RadioCard from './RadioCard/RadioCard';
import Chart from './Chart/Chart';
import './FellowChart.css';

class FellowChart extends Component {
  constructor(props) {
    super(props);
    this.state = {
      selected: 'Trend'
    };
  }

  handleRadioClick = event => {
    const { selected } = this.state;
    const { updateSelected } = this.props;
    this.setState({ selected: event.target.value });
    updateSelected(selected);
  };

  renderRadioCard = () => {
    const { selected } = this.state;
    const { radioCardOptions } = this.props;
    return radioCardOptions.map(entry => (
      <RadioCard
        key={entry.value}
        handleRadioClick={this.handleRadioClick}
        current={selected}
        {...entry}
      />
    ));
  };

  renderCloseButton = () => {
    const { handleChartClose } = this.props;
    return (
      <button
        type="button"
        className="close fellow-chart-close"
        aria-label="Close Fellow Chart"
        onClick={handleChartClose}
      >
        <i className="far fa-times-circle" aria-hidden="true" />
      </button>
    );
  };

  render() {
    const { selected } = this.state;
    const { fellowsCount, radioCardOptions, fellowChartTooltip } = this.props;
    const title = radioCardOptions.find(({ value }) => value === selected).name;
    return (
      <div className="fellow-chart-container">
        <div className="base" style={fellowChartTooltip} />
        <div className="fellow-chart">
          {this.renderCloseButton()}
          {this.renderRadioCard()}
          <Chart data={fellowsCount} title={title} />
        </div>
      </div>
    );
  }
}
FellowChart.defaultProps = {
  fellowChartTooltip: { '--fellow-chart-tooltip': '13%' }
};

FellowChart.propTypes = {
  fellowsCount: PropTypes.arrayOf(PropTypes.object).isRequired,
  radioCardOptions: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleChartClose: PropTypes.func.isRequired,
  updateSelected: PropTypes.func.isRequired,
  fellowChartTooltip: PropTypes.shape({
    '--fellow-chart-tooltip': PropTypes.string.isRequired
  })
};

export default FellowChart;
