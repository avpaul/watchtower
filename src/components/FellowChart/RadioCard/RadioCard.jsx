import React from 'react';
import PropTypes from 'prop-types';
import './RadioCard.css';

const RadioCard = ({ name, count, handleRadioClick, value, current }) => (
  <div className="radio-card form-check form-check-inline">
    <label className="form-check-label" htmlFor={`line-chart-${value}`}>
      <input
        type="radio"
        id={`line-chart-${value}`}
        value={value}
        className="radio-card__input form-check-input"
        onChange={handleRadioClick}
        checked={current === value}
      />
      {`${name} - ${count}`}
    </label>
  </div>
);
RadioCard.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  current: PropTypes.string.isRequired,
  count: PropTypes.number.isRequired,
  handleRadioClick: PropTypes.func.isRequired
};

export default RadioCard;
