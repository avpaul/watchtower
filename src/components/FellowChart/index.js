import FellowChartContainer from './FellowChartContainer';

export default FellowChartContainer;
