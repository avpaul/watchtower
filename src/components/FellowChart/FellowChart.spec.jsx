import React from 'react';
import { shallow } from 'enzyme';

import fellowsCount from '../../__mocks__/fellowsCount.json';
import RadioCard from './RadioCard/RadioCard';

import FellowChart from './FellowChart';

const radioCardOptions = [
  {
    value: 'Trend',
    name: ` Today`
  },
  { value: 'Today', name: 'Trend' }
];
const handleChartCloseSpy = jest.fn();
const setup = propOverrides => {
  const props = {
    fellowsCount,
    radioCardOptions,
    updateSelected: jest.fn(),
    handleChartClose: handleChartCloseSpy,
    ...propOverrides
  };

  const wrapper = shallow(<FellowChart {...props} />);

  return {
    props,
    wrapper
  };
};

it('renders to match snapshot', () => {
  const { wrapper } = setup();
  expect(wrapper).toMatchSnapshot();
});
it('changes the display when the radio button is clicked', () => {
  const { wrapper } = setup();
  const handleRadioClickSpy = jest.spyOn(
    wrapper.instance(),
    'handleRadioClick'
  );
  wrapper
    .instance()
    .handleRadioClick({ target: { value: radioCardOptions[1].value } });
  expect(handleRadioClickSpy).toHaveBeenCalledTimes(1);
});
it('shows the correct number of radio cards', () => {
  const { wrapper } = setup();
  expect(wrapper.find(RadioCard)).toHaveLength(radioCardOptions.length);
});
