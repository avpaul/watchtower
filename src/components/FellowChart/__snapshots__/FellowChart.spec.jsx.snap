// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders to match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FellowChart
    fellowChartTooltip={
        Object {
            "--fellow-chart-tooltip": "13%",
          }
    }
    fellowsCount={
        Array [
            Object {
              "Off Track": 10,
              "On Track": 30,
              "PIP": 3,
              "name": "Week 1",
            },
            Object {
              "Off Track": 12,
              "On Track": 19,
              "PIP": 2,
              "name": "Week 2",
            },
            Object {
              "Off Track": 8,
              "On Track": 22,
              "PIP": 4,
              "name": "Week 3",
            },
            Object {
              "Off Track": 12,
              "On Track": 19,
              "PIP": 5,
              "name": "Week 4",
            },
            Object {
              "Off Track": 4,
              "On Track": 16,
              "PIP": 6,
              "name": "Week 5",
            },
            Object {
              "Off Track": 3,
              "On Track": 23,
              "PIP": 2,
              "name": "Week 6",
            },
            Object {
              "Off Track": 5,
              "On Track": 22,
              "PIP": 1,
              "name": "Week 7",
            },
            Object {
              "Off Track": 9,
              "On Track": 29,
              "PIP": 9,
              "name": "Week 8",
            },
            Object {
              "Off Track": 3,
              "On Track": 30,
              "PIP": 6,
              "name": "Week 9",
            },
            Object {
              "Off Track": 5,
              "On Track": 25,
              "PIP": 0,
              "name": "Week 10",
            },
            Object {
              "Off Track": 6,
              "On Track": 26,
              "PIP": 6,
              "name": "Week 11",
            },
            Object {
              "Off Track": 26,
              "On Track": 30,
              "PIP": 3,
              "name": "Week 12",
            },
          ]
    }
    handleChartClose={[Function]}
    radioCardOptions={
        Array [
            Object {
              "name": " Today",
              "value": "Trend",
            },
            Object {
              "name": "Trend",
              "value": "Today",
            },
          ]
    }
    updateSelected={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="base"
          style={
                    Object {
                              "--fellow-chart-tooltip": "13%",
                            }
          }
/>,
        <div
          className="fellow-chart"
>
          <button
                    aria-label="Close Fellow Chart"
                    className="close fellow-chart-close"
                    onClick={[Function]}
                    type="button"
          >
                    <i
                              aria-hidden="true"
                              className="far fa-times-circle"
                    />
          </button>
          <RadioCard
                    current="Trend"
                    handleRadioClick={[Function]}
                    name=" Today"
                    value="Trend"
          />
          <RadioCard
                    current="Trend"
                    handleRadioClick={[Function]}
                    name="Trend"
                    value="Today"
          />
          <Chart
                    data={
                              Array [
                                        Object {
                                          "Off Track": 10,
                                          "On Track": 30,
                                          "PIP": 3,
                                          "name": "Week 1",
                                        },
                                        Object {
                                          "Off Track": 12,
                                          "On Track": 19,
                                          "PIP": 2,
                                          "name": "Week 2",
                                        },
                                        Object {
                                          "Off Track": 8,
                                          "On Track": 22,
                                          "PIP": 4,
                                          "name": "Week 3",
                                        },
                                        Object {
                                          "Off Track": 12,
                                          "On Track": 19,
                                          "PIP": 5,
                                          "name": "Week 4",
                                        },
                                        Object {
                                          "Off Track": 4,
                                          "On Track": 16,
                                          "PIP": 6,
                                          "name": "Week 5",
                                        },
                                        Object {
                                          "Off Track": 3,
                                          "On Track": 23,
                                          "PIP": 2,
                                          "name": "Week 6",
                                        },
                                        Object {
                                          "Off Track": 5,
                                          "On Track": 22,
                                          "PIP": 1,
                                          "name": "Week 7",
                                        },
                                        Object {
                                          "Off Track": 9,
                                          "On Track": 29,
                                          "PIP": 9,
                                          "name": "Week 8",
                                        },
                                        Object {
                                          "Off Track": 3,
                                          "On Track": 30,
                                          "PIP": 6,
                                          "name": "Week 9",
                                        },
                                        Object {
                                          "Off Track": 5,
                                          "On Track": 25,
                                          "PIP": 0,
                                          "name": "Week 10",
                                        },
                                        Object {
                                          "Off Track": 6,
                                          "On Track": 26,
                                          "PIP": 6,
                                          "name": "Week 11",
                                        },
                                        Object {
                                          "Off Track": 26,
                                          "On Track": 30,
                                          "PIP": 3,
                                          "name": "Week 12",
                                        },
                                      ]
                    }
                    title=" Today"
          />
</div>,
      ],
      "className": "fellow-chart-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "base",
          "style": Object {
            "--fellow-chart-tooltip": "13%",
          },
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              aria-label="Close Fellow Chart"
              className="close fellow-chart-close"
              onClick={[Function]}
              type="button"
>
              <i
                            aria-hidden="true"
                            className="far fa-times-circle"
              />
</button>,
            Array [
              <RadioCard
                current="Trend"
                handleRadioClick={[Function]}
                name=" Today"
                value="Trend"
/>,
              <RadioCard
                current="Trend"
                handleRadioClick={[Function]}
                name="Trend"
                value="Today"
/>,
            ],
            <Chart
              data={
                            Array [
                                          Object {
                                            "Off Track": 10,
                                            "On Track": 30,
                                            "PIP": 3,
                                            "name": "Week 1",
                                          },
                                          Object {
                                            "Off Track": 12,
                                            "On Track": 19,
                                            "PIP": 2,
                                            "name": "Week 2",
                                          },
                                          Object {
                                            "Off Track": 8,
                                            "On Track": 22,
                                            "PIP": 4,
                                            "name": "Week 3",
                                          },
                                          Object {
                                            "Off Track": 12,
                                            "On Track": 19,
                                            "PIP": 5,
                                            "name": "Week 4",
                                          },
                                          Object {
                                            "Off Track": 4,
                                            "On Track": 16,
                                            "PIP": 6,
                                            "name": "Week 5",
                                          },
                                          Object {
                                            "Off Track": 3,
                                            "On Track": 23,
                                            "PIP": 2,
                                            "name": "Week 6",
                                          },
                                          Object {
                                            "Off Track": 5,
                                            "On Track": 22,
                                            "PIP": 1,
                                            "name": "Week 7",
                                          },
                                          Object {
                                            "Off Track": 9,
                                            "On Track": 29,
                                            "PIP": 9,
                                            "name": "Week 8",
                                          },
                                          Object {
                                            "Off Track": 3,
                                            "On Track": 30,
                                            "PIP": 6,
                                            "name": "Week 9",
                                          },
                                          Object {
                                            "Off Track": 5,
                                            "On Track": 25,
                                            "PIP": 0,
                                            "name": "Week 10",
                                          },
                                          Object {
                                            "Off Track": 6,
                                            "On Track": 26,
                                            "PIP": 6,
                                            "name": "Week 11",
                                          },
                                          Object {
                                            "Off Track": 26,
                                            "On Track": 30,
                                            "PIP": 3,
                                            "name": "Week 12",
                                          },
                                        ]
              }
              title=" Today"
/>,
          ],
          "className": "fellow-chart",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "aria-label": "Close Fellow Chart",
              "children": <i
                aria-hidden="true"
                className="far fa-times-circle"
/>,
              "className": "close fellow-chart-close",
              "onClick": [Function],
              "type": "button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-hidden": "true",
                "className": "far fa-times-circle",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": "Trend",
            "nodeType": "function",
            "props": Object {
              "current": "Trend",
              "handleRadioClick": [Function],
              "name": " Today",
              "value": "Trend",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "Today",
            "nodeType": "function",
            "props": Object {
              "current": "Trend",
              "handleRadioClick": [Function],
              "name": "Trend",
              "value": "Today",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "data": Array [
                Object {
                  "Off Track": 10,
                  "On Track": 30,
                  "PIP": 3,
                  "name": "Week 1",
                },
                Object {
                  "Off Track": 12,
                  "On Track": 19,
                  "PIP": 2,
                  "name": "Week 2",
                },
                Object {
                  "Off Track": 8,
                  "On Track": 22,
                  "PIP": 4,
                  "name": "Week 3",
                },
                Object {
                  "Off Track": 12,
                  "On Track": 19,
                  "PIP": 5,
                  "name": "Week 4",
                },
                Object {
                  "Off Track": 4,
                  "On Track": 16,
                  "PIP": 6,
                  "name": "Week 5",
                },
                Object {
                  "Off Track": 3,
                  "On Track": 23,
                  "PIP": 2,
                  "name": "Week 6",
                },
                Object {
                  "Off Track": 5,
                  "On Track": 22,
                  "PIP": 1,
                  "name": "Week 7",
                },
                Object {
                  "Off Track": 9,
                  "On Track": 29,
                  "PIP": 9,
                  "name": "Week 8",
                },
                Object {
                  "Off Track": 3,
                  "On Track": 30,
                  "PIP": 6,
                  "name": "Week 9",
                },
                Object {
                  "Off Track": 5,
                  "On Track": 25,
                  "PIP": 0,
                  "name": "Week 10",
                },
                Object {
                  "Off Track": 6,
                  "On Track": 26,
                  "PIP": 6,
                  "name": "Week 11",
                },
                Object {
                  "Off Track": 26,
                  "On Track": 30,
                  "PIP": 3,
                  "name": "Week 12",
                },
              ],
              "title": " Today",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="base"
            style={
                        Object {
                                    "--fellow-chart-tooltip": "13%",
                                  }
            }
/>,
          <div
            className="fellow-chart"
>
            <button
                        aria-label="Close Fellow Chart"
                        className="close fellow-chart-close"
                        onClick={[Function]}
                        type="button"
            >
                        <i
                                    aria-hidden="true"
                                    className="far fa-times-circle"
                        />
            </button>
            <RadioCard
                        current="Trend"
                        handleRadioClick={[Function]}
                        name=" Today"
                        value="Trend"
            />
            <RadioCard
                        current="Trend"
                        handleRadioClick={[Function]}
                        name="Trend"
                        value="Today"
            />
            <Chart
                        data={
                                    Array [
                                                Object {
                                                  "Off Track": 10,
                                                  "On Track": 30,
                                                  "PIP": 3,
                                                  "name": "Week 1",
                                                },
                                                Object {
                                                  "Off Track": 12,
                                                  "On Track": 19,
                                                  "PIP": 2,
                                                  "name": "Week 2",
                                                },
                                                Object {
                                                  "Off Track": 8,
                                                  "On Track": 22,
                                                  "PIP": 4,
                                                  "name": "Week 3",
                                                },
                                                Object {
                                                  "Off Track": 12,
                                                  "On Track": 19,
                                                  "PIP": 5,
                                                  "name": "Week 4",
                                                },
                                                Object {
                                                  "Off Track": 4,
                                                  "On Track": 16,
                                                  "PIP": 6,
                                                  "name": "Week 5",
                                                },
                                                Object {
                                                  "Off Track": 3,
                                                  "On Track": 23,
                                                  "PIP": 2,
                                                  "name": "Week 6",
                                                },
                                                Object {
                                                  "Off Track": 5,
                                                  "On Track": 22,
                                                  "PIP": 1,
                                                  "name": "Week 7",
                                                },
                                                Object {
                                                  "Off Track": 9,
                                                  "On Track": 29,
                                                  "PIP": 9,
                                                  "name": "Week 8",
                                                },
                                                Object {
                                                  "Off Track": 3,
                                                  "On Track": 30,
                                                  "PIP": 6,
                                                  "name": "Week 9",
                                                },
                                                Object {
                                                  "Off Track": 5,
                                                  "On Track": 25,
                                                  "PIP": 0,
                                                  "name": "Week 10",
                                                },
                                                Object {
                                                  "Off Track": 6,
                                                  "On Track": 26,
                                                  "PIP": 6,
                                                  "name": "Week 11",
                                                },
                                                Object {
                                                  "Off Track": 26,
                                                  "On Track": 30,
                                                  "PIP": 3,
                                                  "name": "Week 12",
                                                },
                                              ]
                        }
                        title=" Today"
            />
</div>,
        ],
        "className": "fellow-chart-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "base",
            "style": Object {
              "--fellow-chart-tooltip": "13%",
            },
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                aria-label="Close Fellow Chart"
                className="close fellow-chart-close"
                onClick={[Function]}
                type="button"
>
                <i
                                aria-hidden="true"
                                className="far fa-times-circle"
                />
</button>,
              Array [
                <RadioCard
                  current="Trend"
                  handleRadioClick={[Function]}
                  name=" Today"
                  value="Trend"
/>,
                <RadioCard
                  current="Trend"
                  handleRadioClick={[Function]}
                  name="Trend"
                  value="Today"
/>,
              ],
              <Chart
                data={
                                Array [
                                                Object {
                                                  "Off Track": 10,
                                                  "On Track": 30,
                                                  "PIP": 3,
                                                  "name": "Week 1",
                                                },
                                                Object {
                                                  "Off Track": 12,
                                                  "On Track": 19,
                                                  "PIP": 2,
                                                  "name": "Week 2",
                                                },
                                                Object {
                                                  "Off Track": 8,
                                                  "On Track": 22,
                                                  "PIP": 4,
                                                  "name": "Week 3",
                                                },
                                                Object {
                                                  "Off Track": 12,
                                                  "On Track": 19,
                                                  "PIP": 5,
                                                  "name": "Week 4",
                                                },
                                                Object {
                                                  "Off Track": 4,
                                                  "On Track": 16,
                                                  "PIP": 6,
                                                  "name": "Week 5",
                                                },
                                                Object {
                                                  "Off Track": 3,
                                                  "On Track": 23,
                                                  "PIP": 2,
                                                  "name": "Week 6",
                                                },
                                                Object {
                                                  "Off Track": 5,
                                                  "On Track": 22,
                                                  "PIP": 1,
                                                  "name": "Week 7",
                                                },
                                                Object {
                                                  "Off Track": 9,
                                                  "On Track": 29,
                                                  "PIP": 9,
                                                  "name": "Week 8",
                                                },
                                                Object {
                                                  "Off Track": 3,
                                                  "On Track": 30,
                                                  "PIP": 6,
                                                  "name": "Week 9",
                                                },
                                                Object {
                                                  "Off Track": 5,
                                                  "On Track": 25,
                                                  "PIP": 0,
                                                  "name": "Week 10",
                                                },
                                                Object {
                                                  "Off Track": 6,
                                                  "On Track": 26,
                                                  "PIP": 6,
                                                  "name": "Week 11",
                                                },
                                                Object {
                                                  "Off Track": 26,
                                                  "On Track": 30,
                                                  "PIP": 3,
                                                  "name": "Week 12",
                                                },
                                              ]
                }
                title=" Today"
/>,
            ],
            "className": "fellow-chart",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "aria-label": "Close Fellow Chart",
                "children": <i
                  aria-hidden="true"
                  className="far fa-times-circle"
/>,
                "className": "close fellow-chart-close",
                "onClick": [Function],
                "type": "button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "aria-hidden": "true",
                  "className": "far fa-times-circle",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": "Trend",
              "nodeType": "function",
              "props": Object {
                "current": "Trend",
                "handleRadioClick": [Function],
                "name": " Today",
                "value": "Trend",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "Today",
              "nodeType": "function",
              "props": Object {
                "current": "Trend",
                "handleRadioClick": [Function],
                "name": "Trend",
                "value": "Today",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "data": Array [
                  Object {
                    "Off Track": 10,
                    "On Track": 30,
                    "PIP": 3,
                    "name": "Week 1",
                  },
                  Object {
                    "Off Track": 12,
                    "On Track": 19,
                    "PIP": 2,
                    "name": "Week 2",
                  },
                  Object {
                    "Off Track": 8,
                    "On Track": 22,
                    "PIP": 4,
                    "name": "Week 3",
                  },
                  Object {
                    "Off Track": 12,
                    "On Track": 19,
                    "PIP": 5,
                    "name": "Week 4",
                  },
                  Object {
                    "Off Track": 4,
                    "On Track": 16,
                    "PIP": 6,
                    "name": "Week 5",
                  },
                  Object {
                    "Off Track": 3,
                    "On Track": 23,
                    "PIP": 2,
                    "name": "Week 6",
                  },
                  Object {
                    "Off Track": 5,
                    "On Track": 22,
                    "PIP": 1,
                    "name": "Week 7",
                  },
                  Object {
                    "Off Track": 9,
                    "On Track": 29,
                    "PIP": 9,
                    "name": "Week 8",
                  },
                  Object {
                    "Off Track": 3,
                    "On Track": 30,
                    "PIP": 6,
                    "name": "Week 9",
                  },
                  Object {
                    "Off Track": 5,
                    "On Track": 25,
                    "PIP": 0,
                    "name": "Week 10",
                  },
                  Object {
                    "Off Track": 6,
                    "On Track": 26,
                    "PIP": 6,
                    "name": "Week 11",
                  },
                  Object {
                    "Off Track": 26,
                    "On Track": 30,
                    "PIP": 3,
                    "name": "Week 12",
                  },
                ],
                "title": " Today",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
